### PR TITLE
improve BNN results

### PR DESCRIPTION
**Pull request by miro-4231**
1. Updated requirements.txt
2. Created an MLP model class
3. Added a new Bnn model that implements Dropout and Batch Normalisation techniques
4. implemented a new train function to take advantage of the new models
5. Adjusted the model parameters
6. added an early stop criterion to the training loop

### CODE REVIEW

**Comment by miro-4231** [comment]
😐

**Comment by 1hachem** [comment]
😃

**Comment by miro-4231** [comment]
Sooooo, could you please tell me how to improve my work??

**Comment by 1hachem** [comment]
- the most important task that needs to be done here is to visualize the uncertainty of the model
- please keep only one training loop in the project (now we have two `train model` and `train_bnn`

**Comment by miro-4231** [comment]
- if i remember correctly, the uncertainty can be visualized in the plot
section
- I really felt reluctant to delete the work done by others 😅
- what do you think about the performance of new models?

**Comment by 1hachem** [comment]
this is a scatter plot :
image

this is what you need to reproduce
image

**Comment by 1hachem** [comment]
don't delete, contribute !

**Comment by miro-4231** [comment]
I don’t think that the dropout method can output such plots

**Comment by miro-4231** [comment]
So i need to change the original?
